### DISCUSSION TITLE
Is Windows 11 supported?

### DISCUSSION THREAD

**Discussion by p-dor** [fohrloop/wakepy#375] [Q&A] [answered]
Hi, I assumed from the documentation that Windows 11 was supported.

However this simple script, running under Python 3.11.5:
```
from wakepy import keep, ActivationResult
import time

def wakePyFailure(iResult):
	print('Failed to keep system awake mode={} active={} success={} err={}'.format(iResult.mode_name, iResult.active_method, iResult.success, iResult.get_failure_text()))

with keep.running(methods = [], on_fail = wakePyFailure):
	print('### LONG TASK ###')
	time.sleep(3)
	print('### EXIT TASK ###')
```
fails with that output:
```
Failed to keep system awake mode=keep.running active=None success=False err=Could not activate Mode "keep.running"!
Method usage results, in order (highest priority first):
[]
### LONG TASK ###
### EXIT TASK ###
```

Any clue on what is going wrong? Any suggestion to try to know more about the error? The get_failure_text() method doesn't really help to understand better...

Thanks!

**Comment by fohrloop** [comment]
Hi @p-dor and thank you for the question. Are you using wakepy 0.9.1? It should definitely support Windows 11 as well as Windows 10. I unfortunately have currently only access to a Windows 10 machine, and I have not yet tested wakepy on Windows 11 myself. If you can help me out a bit I probably can create a patch for you. 

#### A few questions
- Can you confirm the wakepy version you're using? (`import wakepy; print(wakepy.__version__)`)
- How exactly (from which process) you start your python script? Are you using regular CPython on Windows, or WSL/cygwin or something else?
- What platform wakepy thinks you are on?

```
>>> from wakepy.core.platform import CURRENT_PLATFORM
>>> CURRENT_PLATFORM
<PlatformName.LINUX: 'LINUX'>
```

#### About the empty list in the ActivationResult.get_failure_text()

In addition, the fact that you see an empty list (`[]`) in:

```
Method usage results, in order (highest priority first):
[]
```

says that wakepy did not try to use any [wakepy Methods](https://wakepy.readthedocs.io/stable/methods-reference.html) to activate the keep.running mode. That to me would tell that the detected platform for some reason is OTHER (=unknown).

**Comment by p-dor** [comment]
Thanks @fohrloop for your quick answer and happy to help you to find the issue.

Modify my script as below:
```
import wakepy
from wakepy import keep, ActivationResult
from wakepy.core.platform import CURRENT_PLATFORM
import time

def wakePyFailure(iResult):
	print('Failed to keep system awake mode={} active={} success={} err={}'.format(iResult.mode_name, iResult.active_method, iResult.success, iResult.get_failure_text()))

print('wakepy version -> {}'.format(wakepy.__version__))
print('Current platform -> {}'.format(CURRENT_PLATFORM))

with keep.running(methods = [], on_fail = wakePyFailure):
	print('### LONG TASK ###')
	time.sleep(3)
	print('### EXIT TASK ###')
```

This is the output:

```
wakepy version -> 0.9.1
Current platform -> WINDOWS
Failed to keep system awake mode=keep.running active=None success=False err=Could not activate Mode "keep.running"!

Method usage results, in order (highest priority first):
[]
### LONG TASK ###
### EXIT TASK ###
```

I'm using regular Python 3.11.5 for Windows (provided [here](https://www.python.org/downloads/windows/)) and I'm executing the script directly from a Windows PowerShell terminal.

Thanks!

**Reply by fohrloop** [reply, marked as answer]
Thank you for the information. Seems that I overlooked your code a bit. You have given `methods = []`, which essentially says to wakepy to not use _any_ methods to try to enter the mode. This will force a failure. Please see documentation for the [keep.running](https://wakepy.readthedocs.io/stable/api-reference.html#wakepy.keep.running) mode:

> methods: list, tuple or set of str
The names of Methods to select from the keep.running mode; a “whitelist” filter. Means “use these and only these Methods”. ...

and the note in the [Example: Notify user with a custom callable](https://wakepy.readthedocs.io/stable/user-guide.html#example-notify-user-with-a-custom-callable):

![image](https://github.com/user-attachments/assets/669ec8a8-5739-4550-82b9-7c91553a4641)

The fix is simply _not_ provide the `methods` argument at all, if you do not have any special requirements for controlling the activation process. I.e, just use:

```python
with keep.running(on_fail = wakePyFailure):
	print('### LONG TASK ###')
	time.sleep(3)
	print('### EXIT TASK ###')
```

**Comment by p-dor** [comment]
Ooops sorry, my bad, I copied your sample to catch error without looking at that param which was obviously to force the error...
I confirm it works fine with your suggested fix.
Thanks again for your help!
